### PR TITLE
tests: implement unit test for transactions that initialize scripts

### PR DESCRIPTION
**Pull request by LGLO**
# Description

Adds transaction unit tests.

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages.
- [x] New tests are added if needed and existing tests are updated.
- [ ] Relevant logging and metrics added
- [ ] CI passes. See note on CI.
- [ ] Any changes are noted in the `changelog.md` for affected crate
- [x] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG Partner Chains developers to do this
for you.

### CODE REVIEW

File: toolkit/offchain/src/reserve/init.rs [hunk 1]
```diff
@@ -314,3 +312,219 @@ fn decode_version_oracle_validator_datum(data: PlutusData) -> Option<VersionOrac
...
+					.script_hash()
+					.to_vec(),
+			));
+			let constr_plutus_data = ConstrPlutusData::new(&BigNum::one(), &list);
+			PlutusData::new_constr_plutus_data(&constr_plutus_data)
+		};
+
+		let _ = redeemers
+			.clone()
+			.into_iter()
```
Comment (kpinter-iohk) [inline toolkit/offchain/src/reserve/init.rs:453]:
```suggestion
			.iter()
```

File: toolkit/offchain/src/reserve/init.rs [hunk 2]
```diff
@@ -314,3 +312,219 @@ fn decode_version_oracle_validator_datum(data: PlutusData) -> Option<VersionOrac
...
+			.find(|r| {
+				r.tag() == RedeemerTag::new_mint()
+					&& r.data() == expected_vo_redeemer_data
+					&& r.ex_units() == versioning_script_cost()
+			})
+			.expect("Transaction should have a mint redeemer for Version Oracle Policy");
+		let _ = redeemers
+			.into_iter()
```
Comment (kpinter-iohk) [inline toolkit/offchain/src/reserve/init.rs:461]:
```suggestion
			.iter()
```

**Review by kpinter-iohk — APPROVED** [review]
LGTM, minor comments

File: toolkit/offchain/src/csl.rs [hunk 3]
```diff
@@ -246,6 +246,16 @@ impl OgmiosUtxoExt for OgmiosUtxo {
 	}
 }
 
+pub trait UtxoIdExt {
+	fn to_csl(&self) -> TransactionInput;
+}
+
+impl UtxoIdExt for UtxoId {
```
[the remark this answers is not in the document]
Comment (LGLO) [reply toolkit/offchain/src/csl.rs:253]: Yes, I'd like to keep `domain` free from CSL dependency.

**Pull request merged**